cmd/aws/create_test.go

```diff
@@ -0,0 +1,402 @@
+package aws
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	ec2Types "github.com/aws/aws-sdk-go-v2/service/ec2/types"
+	"github.com/aws/aws-sdk-go-v2/service/ssm"
+	ssmTypes "github.com/aws/aws-sdk-go-v2/service/ssm/types"
+	"github.com/stretchr/testify/assert"
+)
+
+type mockCredentialsProvider struct {
+	creds aws.Credentials
+	err   error
+}
+
+func (m *mockCredentialsProvider) Retrieve(ctx context.Context) (aws.Credentials, error) {
+	return m.creds, m.err
+}
+
+func TestValidateCredentials(t *testing.T) {
+	tests := []struct {
+		name        string
+		creds       aws.Credentials
+		err         error
+		expectedErr error
+	}{
+		{
+			name: "valid credentials",
+			creds: aws.Credentials{
+				AccessKeyID:     "test-access-key-id",
+				SecretAccessKey: "test-secret-access-key",
+				SessionToken:    "test-session-token",
+			},
+			err:         nil,
+			expectedErr: nil,
+		},
+		{
+			name:        "failed to retrieve credentials",
+			creds:       aws.Credentials{},
+			err:         errors.New("failed to retrieve credentials"),
+			expectedErr: errors.New("failed to retrieve AWS credentials: failed to retrieve credentials"),
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			mockProvider := &mockCredentialsProvider{
+				creds: tt.creds,
+				err:   tt.err,
+			}
```

Comment:
I see your intention here, but it's more recommendable to change the behaviour of the mock to dynamically set the function rather than its return values. 

I would recommend you to change the mock structure above to be:

```go
type mockCredentialsProvider struct {
	fnRetrieve (ctx context.Context) (aws.Credentials, error)
}

func (m *mockCredentialsProvider) Retrieve(ctx context.Context) (aws.Credentials, error) {
	if m.fnRetrieve == nil {
		return errors.New("not implemented")
	}

	return m.fnRetrieve(ctx)
}
```

Then each test can decide what `Retrieve()` looks like and return, rather than dynamically handling its error messages and returned values. 

The other reason is that this makes the mock usable just once before reinitialization if you want to test in the same test multiple case scenarios. You often don't want to go that way.